src/test/java/seedu/address/logic/parser/EditCommandParserTest.java

```
@@ -0,0 +1,215 @@
+        // "q/unknown is this k/parsed" is parsed as Name, which results in an error as well.
+        String invalidPrefix = "q/";
+        assertParseFailure("1 " + invalidPrefix + "unknown", Name.MESSAGE_NAME_CONSTRAINTS);
+    }
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(VALID_NAME_AMY, MESSAGE_INVALID_FORMAT);
+
+        // no field specified
+        assertParseFailure("1", EditCommand.MESSAGE_NOT_EDITED);
+
+        // no index and no field specified
+        assertParseFailure("", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidIndex_failure() {
+        // negative index
+        assertParseFailure("-5" + NAME_DESC_AMY, MESSAGE_INVALID_FORMAT);
+
+        // zero
+        assertParseFailure("0" + NAME_DESC_AMY, MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_singleInvalidValue_failure() {
+        // invalid name
+        assertParseFailure("1" + INVALID_NAME_DESC, Name.MESSAGE_NAME_CONSTRAINTS);
+
+        // invalid phone
+        assertParseFailure("1" + INVALID_PHONE_DESC, Phone.MESSAGE_PHONE_CONSTRAINTS);
+
+        // invalid email
+        assertParseFailure("1" + INVALID_EMAIL_DESC, Email.MESSAGE_EMAIL_CONSTRAINTS);
+
+        // invalid address
+        assertParseFailure("1" + INVALID_ADDRESS_DESC, Address.MESSAGE_ADDRESS_CONSTRAINTS);
+
+        // invalid tag
+        assertParseFailure("1" + INVALID_TAG_DESC, Tag.MESSAGE_TAG_CONSTRAINTS);
+    }
+
+    @Test
+    public void parse_singleInvalidValueWithOtherValidValues_failure() {
+        // invalid phone followed by valid email
+        assertParseFailure("1" + INVALID_PHONE_DESC + EMAIL_DESC_AMY, Phone.MESSAGE_PHONE_CONSTRAINTS);
+
+        // valid phone followed by invalid phone
+        assertParseFailure("1" + PHONE_DESC_BOB + INVALID_PHONE_DESC, Phone.MESSAGE_PHONE_CONSTRAINTS);
```

Upon testing for this test case, I realised that our parser actually ignores the error in the first value.